crates/tako/src/gateway.rs

```diff
@@ -42,12 +42,32 @@ impl Default for ResourceRequest {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone)]
+pub struct ResourceRequestVariants {
+    pub variants: SmallVec<[ResourceRequest; 1]>,
+}
+
+impl ResourceRequestVariants {
+    pub fn min_time(&self) -> Duration {
+        self.variants
+            .iter()
+            .map(|rq| rq.min_time)
+            .min()
```

Comment:
I wonder if this shouldn't be max(), at least for autoalloc usage.

Reply:
It depends how autoalloc is using it. What about requests (e.g. for gpus) that needs 10mins OR 1h (for cpus). Then it should be runnable on gpu queue that has timelimit >10min.